server/clients/pokemon-client.js

```diff
@@ -3,11 +3,18 @@ const axios = require("axios").default;
 class PokemonClient {
    constructor() {
       this.API_URL = "https://pokeapi.co/api/v2/pokemon/";
+      this.cache = [];
    }
 
    async getPokemon(pokemonID) {
       try {
+         // Cache holds previous fetched pokemons data. If id does not exist in cache then fetch from api
+         if (this.cache.some((e) => e.id === pokemonID)) {
```

Comment:
I would recomment making the cache an object instead of array.
when it's array, you need to scan all the items in order to look for the item with the correct id (n times)
if you convert it to object, you could save the items by id and access it in o(1) (without scanning all the items)
{
   1: {id:1, data: {}},
   2: {id:2, data:{}}
}

in js we use object to achieve this but this concept in represented by **hashtable**, which I really recommend reading of.

Reply:
Oh nice! I know a little bit about hashtables but never used them before. I just reimplemented this with Map()